internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -1238,6 +1238,10 @@ components:
             there are one or more mountpoints in which case it will use LVM. 'lvm' always
             uses LVM, even when there are no extra mountpoints. 'raw' uses raw partitions
             even when there are one or more mountpoints.
+        fips:
+          type: boolean
+          description: |
+            Enables the system FIPS mode
```

Comment:
This goes for both the blueprint and the cloud API:
Is it possible we'll ever need more fips-related options?  I'm not familiar so I can't say but if there's ever any need to, say, define a mode or a mode option or whatever, this will be a bit awkward to work with and it would be better if we made it:

```json
"fips": {
  "enabled": true
}
```

so that in the future we can extend it to:

```json
"fips": {
  "enabled": true,
  "some-option": "some-value"
}
```

Reply:
Interesting. I think that the only additional customization we might want to allow in the future related to FIPS would be the crypto policy to apply. This would be something like `"crypto-policy": "FIPS:SUBPOLICY1:SUBPOLICY2..."`. However this is not something exclusive to FIPS but a more general configuration so I would not put it under "fips" anyway.

Reply:
Right.  In that case let's keep it as is... unless it makes more sense to add this new option as `crypto-policy`, make it an enum, and only support `"fips"` as a value for now. :thinking:

Reply:
> Right. In that case let's keep it as is... unless it makes more sense to add this new option as `crypto-policy`, make it an enum, and only support `"fips"` as a value for now. 🤔\

Oh, no no, the crypto policy is related to FIPS but it's not enough to setup the FIPS mode. To enable the FIPS mode more things than setting up the crypto policy to FIPS are needed, like adding the `fips=1` kernel argument or making sure the `fips` dracut module is included in the `initramfs`.

The only customizable thing for FIPS would be using a custom crypto policy but for keeping the system FIPS compliant that custom policy must be based on FIPS default crypto policy and restrict even more the allowed algorithms.

On the other hand, we might want to allow the possibility of setting a custom crypto policy but without using FIPS mode.

Reply:
> Oh, no no, the crypto policy is related to FIPS but it's not enough to setup the FIPS mode. To enable the FIPS mode more things than setting up the crypto policy to FIPS are needed, like adding the `fips=1` kernel argument or making sure the `fips` dracut module is included in the `initramfs`.

Right but as a high level option this could work, no?  I mean we rarely put an option in the API that goes straight down to an option in an osbuild stage without affecting other things.  What I was thinking was that this crypto policy option (or maybe something by a different name that doesn't imply just the one thing) would now take one value, "fips", which enables all the fips-related changes (the same way setting `fips=true` makes all the necessary changes now).  Any other _related_ but _alternative_ value, say a custom policy or a different standard, would be mutually exclusive with fips (so the enum is taking care of this logic) and have its own set of changes, like kernel args, dracut modules etc.

A more general question to help me understand: Is FIPS a unique enough option to be a top level on/off switch in the compose request, or is it one option of potential future alternatives that we might want to support?  and on top of that, is there any chance either FIPS or one of these alternatives will have an associated configuration?

Like say a FIPSv2 (or whatever, I don't know how this particular standard gets set and if it can have revisions) comes along that does things slightly differently and we want to support both, we could have:
```json
{
  "fips": {
    "enabled": true
  }
}
```
and in the future, if needed
```json
{
  "fips": {
    "enabled": true,
    "version": 2
  }
}
```

or, with an enum
```json
{
  "secutiy-standard": "fips"
}
```

and in the future, if needed
```json
{
  "secutiy-standard": "fips|fips_v2|custom|whatever"
}
```

Am I thinking about this wrong?

This saves us from needing to have a top-level on/off switch that conflicts with a potential new option, for example, which makes the API have fewer interdependencies between options and less confusing (IMO at least).